test/util/testnode/config.go

```diff
@@ -0,0 +1,163 @@
+package testnode
+
+import (
+	"time"
+
+	"github.com/celestiaorg/celestia-app/cmd/celestia-appd/cmd"
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	pruningtypes "github.com/cosmos/cosmos-sdk/pruning/types"
+	"github.com/cosmos/cosmos-sdk/server"
+	srvconfig "github.com/cosmos/cosmos-sdk/server/config"
+	srvtypes "github.com/cosmos/cosmos-sdk/server/types"
+	"github.com/tendermint/tendermint/config"
+	tmconfig "github.com/tendermint/tendermint/config"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"github.com/tendermint/tendermint/types"
+)
+
+// Config is the configuration of a test node.
+type Config struct {
+	// ChainID is the chain ID of the network.
+	ChainID string
+	// TmConfig is the Tendermint configuration used for the network.
+	TmConfig *tmconfig.Config
+	// AppConfig is the application configuration of the test node.
+	AppConfig *srvconfig.Config
+	// ConsensusParams are the consensus parameters of the test node.
+	ConsensusParams *tmproto.ConsensusParams
+	// AppOptions are the application options of the test node.
+	AppOptions *KVAppOptions
+	// GenesisOptions are the genesis options of the test node.
+	GenesisOptions []GenesisOption
+	// Accounts are the accounts of the test node.
+	Accounts []string
+	// AppCreator is used to create the application for the testnode.
+	AppCreator srvtypes.AppCreator
```

Comment:
per #1990, we can now replace this function with a new one that creates a malicious applicaiton